### PR TITLE
Support Deepseek w4a8 quantization

### PR DESCRIPTION
**Pull request by pichangping**

### CODE REVIEW

**Comment by wangxiyuan** [comment]
Thanks for your contribution. I have some question before deep review:
1. how to convert a w4a8 weight? which version of modelslim should be used?
2. does deepseek v2 lite works with w4a8 as well? If yes, the e2e test should be added as well. @22dimensions please help to update a weight to modelscope as well.
3. does w4a8 only works with deepseek, how about any other model, like Qwen?

**Comment by pichangping** [comment]
1. First, quantize using the BF16 format to int8, then quantize the obtained int8 to int4. Since there is no int4 data format, the tool outputs the quantized weights using the low-precision int8.
2.Whether it is compatible is not clear
3.Other model features are supported, but accuracy is not guaranteed.

**Comment by pichangping** [comment]
In the main branch of this repository "https://gitee.com/ascend/msit", you can access the file "msmodelslim/example/DeepSeek/quant_deepseek_w4a8.py". Use the command "python quant_deepseek_w4a8.py --model_path DeepSeek-R1-BF16 --save_path DeepSeek-R1-W4A8 --mindie_format". If you are unfamiliar with the parameters, you can read msmodelslim/example/DeepSeek/README.md.

**Comment by pichangping** [comment]
modelslim
Use branch master
Command: Refer to [README.md](https://gitee.com/ascend/msit/blob/master/msmodelslim/example/DeepSeek/README.md) for [运行前必检](https://gitee.com/ascend/msit/blob/master/msmodelslim/example/DeepSeek/README.md#%E8%BF%90%E8%A1%8C%E5%89%8D%E5%BF%85%E6%A3%80) and [DeepSeek-R1 w4a8 混合量化](https://gitee.com/ascend/msit/blob/master/msmodelslim/example/DeepSeek/README.md#deepseek-r1-w4a8-%E6%B7%B7%E5%90%88%E9%87%8F%E5%8C%96%E5%89%8D%E4%B8%89%E5%B1%82-mlpw8a8-dynamic-%E9%87%8F%E5%8C%96mla%E5%85%B1%E4%BA%AB%E4%B8%93%E5%AE%B6w8a8%E9%87%8F%E5%8C%96%E8%B7%AF%E7%94%B1%E4%B8%93%E5%AE%B6w4a8-dynamic%E9%87%8F%E5%8C%96)
Reference command: python3 quant_deepseek_w4a8.py --model_path {float weight path} --save_path {W4A8 quantized weight path} --mindie_format
Since mindie_format generates a mindie format, some adaptation modifications are needed for vllm usage:
Rename quant_model_description_w8a8_dynamic.json to quant_model_description.json, and add the configuration "group_size": 256 in this file
Modify config.json: Change model_type to deepseek_v3; delete quantization_config

File: tests/multicard/test_model_DeepSeek_W4A8,py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#
```
Comment (wangxiyuan) [inline tests/multicard/test_model_DeepSeek_W4A8,py:1]: rebase to main please, and move this file to tests/e2e/multicard

File: vllm_ascend/quantization/w4a8_dynamic.py [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+import torch_npu
+from vllm.distributed import GroupCoordinator
+
+import vllm_ascend.envs as envs_ascend
+from vllm_ascend.distributed.parallel_state import get_ep_group
+from vllm_ascend.ops.fused_moe import select_experts
+from vllm_ascend.utils import dispose_tensor
+from vllm.config import get_current_vllm_config
+
+VLLM_ENABLE_MC2: bool = envs_ascend.VLLM_ENABLE_MC2
```
Comment (Yikun) [inline vllm_ascend/quantization/w4a8_dynamic.py]: Please remove this, this already remove in https://github.com/vllm-project/vllm-ascend/pull/1229

File: tests/multicard/test_model_DeepSeek_W4A8,py (hunk 1 above)
Comment (Yikun) [reply tests/multicard/test_model_DeepSeek_W4A8,py:1]: The file name was also wrong....
